UE/Application/Ports/TimerPort.cpp

```diff
@@ -22,11 +22,28 @@ void TimerPort::stop()
 void TimerPort::startTimer(Duration duration)
 {
     logger.logDebug("Start timer: ", duration.count(), "ms");
+    this->running = true;
+    this->timer = std::async(&TimerPort::timerThread, this, duration);
 }
 
 void TimerPort::stopTimer()
 {
     logger.logDebug("Stop timer");
+    this->running = false;
+    this->timer.get();
+}
+
+void TimerPort::timerThread(Duration duration)
+{
+    using namespace std::chrono_literals;
+    for(int i = duration / 100ms; i > -1; i--)
+    {
+        std::this_thread::sleep_for(100ms);
+        if (!this->running)
```

Comment:
Apart from `this`, we can also use `not` instead of `!`.
```suggestion
        if (not running)
```